### PR TITLE
Upgrade to support new Typescript, vite, Vitest and NX tooling

### PR DESCRIPTION
**Pull request by SimeonC**
# Version

Published prerelease version: `@tablecheck/nx@1.0.1-next.18`

<details>
  <summary>Changelog</summary>

  #### ⚠️ Pushed to `next`
  
  - ci: revert switch next to canary releases due to lerna issue ([@SimeonC](https://github.com/SimeonC))
  - ci: switch next to canary releases due to lerna issue ([@SimeonC](https://github.com/SimeonC))
  - fix: add forcePublish to npm publish to bump all packages in next ([@SimeonC](https://github.com/SimeonC))
  - ci: skip husky in release actions install ([@SimeonC](https://github.com/SimeonC))
  - fix: tests and github_actions ([@SimeonC](https://github.com/SimeonC))
  - ci: fix github actions to have main branch for nx ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/nx`
    - fix: quality generator fixes and cleanups ([@SimeonC](https://github.com/SimeonC))
    - fix(nx): quality executor should run prettier from root cwd to correctly resolve configs and ignores ([@SimeonC](https://github.com/SimeonC))
    - fix: quality generator and migration include file-types ([@SimeonC](https://github.com/SimeonC))
    - fix: add nx meta files to bundle for detection ([@SimeonC](https://github.com/SimeonC))
    - ci: fix github release action ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/commitlint-config`
    - fix: remove start-case restriction in scope-case ([@SimeonC](https://github.com/SimeonC))
    - fix(commitlint-config): use correct junit format via package ([@SimeonC](https://github.com/SimeonC))
    - feat(commitlint-config): add junit formatter ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/frontend-audit`, `@tablecheck/commitlint-config`, `@tablecheck/eslint-config`, `@tablecheck/eslint-plugin`, `@tablecheck/nx`, `@tablecheck/semantic-release-config`, `@tablecheck/frontend-utils`
    - fix: prettier works with v2/v3 though v3 is preferred ([@SimeonC](https://github.com/SimeonC))
    - fix: remove audit project and correct paths for nx schemas ([@SimeonC](https://github.com/SimeonC))
    - feat: full build and passing lint via nx!! ([@SimeonC](https://github.com/SimeonC))
    - feat: build runs and quality packages function ([@SimeonC](https://github.com/SimeonC))
    - feat: move everything except quality ([@SimeonC](https://github.com/SimeonC))
    - feat: scripts is dead, long live scripts! WIP! ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/nx`, `@tablecheck/frontend-utils`
    - feat: remove library build (use nx rollup) fixes for monorepo ([@SimeonC](https://github.com/SimeonC))
    - fix: quality generator bugfixes ([@SimeonC](https://github.com/SimeonC))
    - fix(nx): quality script should run correctly ([@SimeonC](https://github.com/SimeonC))
    - fix: nx quality scripts and prettier fixes ([@SimeonC](https://github.com/SimeonC))
    - fix: esm/cjs interop fixes ([@SimeonC](https://github.com/SimeonC))
    - fix: pre-commit scripts and apply formatting ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/eslint-config`, `@tablecheck/eslint-plugin`, `@tablecheck/nx`, `@tablecheck/prettier-config`, `@tablecheck/frontend-utils`
    - feat: bump prettier and add lib publishing checks ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/eslint-config`, `@tablecheck/eslint-plugin`, `@tablecheck/nx`, `@tablecheck/frontend-utils`
    - fix: assorted bugfixes and rule tweaks ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/frontend-audit`, `@tablecheck/eslint-config`, `@tablecheck/eslint-plugin`, `@tablecheck/nx`, `@tablecheck/frontend-utils`
    - feat: upgrade eslint dependencies ([@SimeonC](https://github.com/SimeonC))
    - fix: bump package versions in deps ([@SimeonC](https://github.com/SimeonC))
    - fix: install from public npm and remove dist files ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/eslint-plugin`, `@tablecheck/nx`, `@tablecheck/frontend-utils`
    - fix: bump packages and deps ([@SimeonC](https://github.com/SimeonC))
    - feat: add stylelint ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/eslint-plugin`, `@tablecheck/nx`, `@tablecheck/prettier-config`, `@tablecheck/frontend-utils`
    - fix: downgrade prettier to v2 ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/frontend-audit`, `@tablecheck/nx`, `@tablecheck/frontend-utils`
    - fix: nx requires commonjs or weird node memory errors occur ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/frontend-audit`, `@tablecheck/nx`, `@tablecheck/prettier-config`
    - fix: migrate script updates and remove dead packages ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/commitlint-config`, `@tablecheck/eslint-config`, `@tablecheck/eslint-plugin`, `@tablecheck/nx`, `@tablecheck/prettier-config`, `@tablecheck/semantic-release-config`, `@tablecheck/frontend-utils`
    - feat: add prettier config file and bump prettier version ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/frontend-audit`, `@tablecheck/commitlint-config`, `@tablecheck/eslint-config`, `@tablecheck/eslint-plugin`, `@tablecheck/nx`, `@tablecheck/semantic-release-config`
    - fix: audit works and passes ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/frontend-audit`, `@tablecheck/eslint-plugin`, `@tablecheck/frontend-utils`
    - fix: import esm/cjs bugs ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/eslint-config`, `@tablecheck/nx`, `@tablecheck/frontend-utils`
    - feat: remove stylelint and fix lint/ts issues ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/eslint-plugin`
    - feat: add shortest import eslint rule ([@SimeonC](https://github.com/SimeonC))
  - `@tablecheck/frontend-audit`, `@tablecheck/commitlint-config`, `@tablecheck/eslint-config`, `@tablecheck/eslint-plugin`, `@tablecheck/semantic-release-config`, `@tablecheck/frontend-utils`
    - chore: deps upgrade ([@SimeonC](https://github.com/SimeonC))
    - feat: major upgrade ([@SimeonC](https://github.com/SimeonC))
    - refactor: get the ts to js build working nicely ([@SimeonC](https://github.com/SimeonC))
  
  #### Authors: 1
  
  - Simeon Cheeseman ([@SimeonC](https://github.com/SimeonC))
</details>

### CODE REVIEW

**Comment by SimeonC** [comment]
TODO: upgrade to the new version of typescript-eslint

**Comment by SimeonC** [comment]
Upgrade Process (Adapted from [Here](https://nx.dev/getting-started/tutorials/react-monorepo-tutorial) and [Here](https://nx.dev/recipes/adopting-nx/manual))

Reference PRs:
* [Booking Form](https://github.com/tablecheck/booking-form/pull/1090)

1. Create an NX project next to the project you want to upgrade (consult the above links, recommend react-monorepo) CleanShot 2023-09-28 at 11 11 00
2. Upgrade Typescript + Vite + Prettier + React to the latest versions
3. Install TC NX packages - `npm i @tablecheck/nx` (will probably need `--legacy-peer-deps`)
4. Run `npm install`
5. Run `npx nx g @tablecheck/nx:quality` and follow the prompts for each "project"
6. Move over the `.npmrc` file from old repo - bump to node >=v20
7. Add a `.nvmrc` with the following; (remove the legacy-peer-deps if possible)
  ```
  registry=https://tablecheck-934763610305.d.codeartifact.ap-northeast-1.amazonaws.com/npm/tablecheck/
  legacy-peer-deps=true
  ```
9. Remove `.prettierrc` (`prettierrc.json` is now used)

File: .gitignore [hunk 1]
```diff
@@ -23,3 +22,9 @@ junit
 .env.test.local
 .env.production.local
 .@tablecheck
+/Users/simeon.cheeseman/Development/tablecheck-react-system/.@tablecheck
```
Comment (SashaShostyr) [inline .gitignore:25]: Wrong path I guess

Comment (SimeonC) [reply .gitignore:25]: Thanks, that shouldn't have been there anyway.

**Comment by SashaShostyr** [comment]
`packages/eslint-plugin/__tests__` contains empty files. Do we need it?

**Pull request merged**